### ISSUE TITLE
Use tabs for API info on docs pages

### ISSUE THREAD

**Issue by phlax** [envoyproxy/envoy#16329]
### description

atm there are various appendages to API info - security status, canonical name, config, extension info

we are also going to add canonical uri, examples, formerly known as info, lifecycle status, and richer extension info

atm it uses `.. notes`, and `.. tips` and in some places `warnings` in others - this is making for a bit of an angry fruit salad

i think we need to shift this info into tabs for API files and fields and figure out how best to prioritize

this will also allow extension examples which adds significantly to the complexity here

**Comment by phlax** [comment]
cc @htuch @mattklein123

**Comment by phlax** [comment]
related #16328 #15384 #15344 #16303

**Comment by htuch** [comment]
Yes, but security and other key warnings must still appear clearly at the top; we can't bury these into tabs because they are unsightly. They are callouts for a reason :)

**Comment by phlax** [comment]
> ...warnings must still appear clearly at the top ... we can't bury these 

yep. i will defo follow the principle of no buried warnings
